server/routes/referrals/referralFormPresenter.ts

```diff
@@ -73,41 +87,83 @@ export default class ReferralFormPresenter {
         type: 'single',
         title: 'Check your answers',
         number: '4',
-        status: this.canSubmitReferral ? ReferralFormStatus.NotStarted : ReferralFormStatus.CannotStartYet,
+        status: checkAnswersSectionFormStatus,
         tasks: [
           {
             title: 'Check your answers',
-            url: this.canSubmitReferral ? 'check-answers' : null,
+            url: checkAnswersSectionFormStatus !== ReferralFormStatus.CannotStartYet ? 'check-answers' : null,
           },
         ],
       },
     ]
   }
 
-  private determineInterventionDetailsSectionStatus(): ReferralFormStatus {
-    const hasCompletedSection = [
+  private referralFormStatus(
+    draftReferralValues: DraftReferralValues,
+    previousFormStatus: ReferralFormStatus = ReferralFormStatus.Completed
+  ): ReferralFormStatus {
+    if (previousFormStatus !== ReferralFormStatus.Completed) {
+      return ReferralFormStatus.CannotStartYet
+    }
+    if (
+      draftReferralValues.every(field => {
```

Comment:
I'm finding the multiline `if` condition a bit tricky to parse here, could it be clearer to assign the value to a variable and use it here?

Reply:
Yes. You are right, it is a bit too messy.
I've fixed it as you've suggested.